developer-docs/bolt_server.md

```diff
@@ -47,11 +109,79 @@ The Target be a JSON object. The following keys are available:
 
+### WinRM Target Object
+The Target be a JSON object. The following keys are available:
+- `hostname`: String, *required* - Target identifier.
+- `password`: String, *optional* - Password for WinRM transport authentication.
+- `port`: Integer, *optional* - Connection port (Default: `5986`, or `5985` if `ssl: false`.)
+- `user`: String, *optional* - Login user (Default: `root`).
+- `connect-timeout`: Integer, *optional* - How long Bolt should wait when establishing connections.
+- `tmpdir`: String, *optional* - The directory to upload and execute temporary files on the target.
+- `ssl`: Boolean, *optional* - When true, Bolt will use normal http connections for WinRM (Default: `true`).
+- `ssl-verify`: Boolean, *optional* - When true, verifies the targets certificate matches the cacert (Default: `true`)
+- `tmpdir`: String, *optional* - The directory to upload and execute temporary files on the target.
+- `cacert`: String, *optional* - The path to the CA certificate.
+- `extensions`: List, *optional* - List of file extensions that are accepted for scripts or tasks. Scripts with these file extensions rely on the target node's file type association to run. For example, if Python is installed on the system, a .py script should run with python.exe. The extensions .ps1, .rb, and .pp are always allowed and run via hard-coded executables
+
 ### Task Object
-The Task be a JSON object. The following keys are available:
-- `name`: String, *required* - The name of the task.
-- `metadata`: String, *optional* - The contents of the task's metadata.json file. 
+This is nearly identical to the [task detail JSON
+object](https://github.com/puppetlabs/puppetserver/blob/master/documentation/puppet-api/v3/task_detail.json)
+from [puppetserver](https://github.com/puppetlabs/puppetserver), with an
+additional `file_content` key.
+
+The task is a JSON object which includes the following keys:
+
+#### Name
+
+The name of the task
+
+#### Metadata
+The metadata object is optional, and contains metadata about the task being run. It includes the following keys:
+
+- `description`: String, *optional* - The task description from it's metadata
+- `parameters`: Object, *optional* - A JSON object whose keys are parameter names, and whose values are JSON objects with 2 keys:
+    - `description`: String, *optional* - The parameter description
+    - `type`: String, *optional* - The type the parameter should accept
+
+#### File
+**NOTE**: We plan to eventually get this information directly from puppetserver
+
+The task file and it's metadata. This is a JSON object that includes the following keys:
+- `filename`: String, *required* - The name of the file the task is in
 - `file_content`: String, *required* - Task's base64 encoded file content.
+- `sha256`: String, *optional* - The SHA of the task file
+- `size_bytes`: String, *optional* - The size of the file
+- `uri`: Object, *optional* - A JSON object with the following keys:
```

Comment:
I'm a little confused by this. Is this for once bolt-server is getting task information directly from puppetserver? Is this currently supported at all?

Reply:
No, not currently supported. I can remove it given that, but left them in this PR in case we did want to make any of them required (ie. check for file size or verify the sha).

Reply:
Maybe just denote that they aren't yet supported?

Reply:
If Bolt's not using them, I think I'd remove them.

Reply:
We can remove these keys from the docs but I think we need to work towards a standard format for task details which similar to what puppet-server provides. Bolt-server shouldn't error if these keys are present.

Reply:
M'k, removed!